node/src/reactor/joiner.rs

```diff
@@ -967,14 +1033,30 @@ impl Reactor {
             .verifiable_chunked_hash_activation;
         match message {
             NetResponse::Deploy(ref serialized_item) => {
-                reactor::handle_fetch_response::<Self, Deploy>(
-                    self,
-                    effect_builder,
-                    rng,
-                    sender,
-                    serialized_item,
-                    verifiable_chunked_hash_activation,
-                )
+                let deploy: Box<Deploy> = match bincode::deserialize::<
+                    FetchedOrNotFound<Deploy, DeployHash>,
+                >(serialized_item)
+                {
+                    Ok(FetchedOrNotFound::Fetched(deploy)) => Box::new(deploy),
+                    Ok(FetchedOrNotFound::NotFound(deploy_hash)) => {
+                        error!(
+                            "peer did not have deploy with hash {}: {}",
+                            sender, deploy_hash
+                        );
+                        return Effects::new();
+                    }
+                    Err(error) => {
+                        error!("failed to decode deploy from {}: {}", sender, error);
+                        return Effects::new();
+                    }
+                };
+
+                let event = JoinerEvent::DeployAcceptor(deploy_acceptor::Event::Accept {
+                    deploy,
+                    source: Source::Peer(sender),
+                    maybe_responder: None,
+                });
+                <Reactor as reactor::Reactor>::dispatch_event(self, effect_builder, rng, event)
```

Comment:
Does `self.dispatch_event` not work here?

Reply:
No, because the `Reactor` trait is not in scope, and importing it would cause a collision with the `Reactor` struct 🤦‍♂️ It works in participating, because there all calls to it are inside the `impl reactor::Reactor for Reactor` block...

Reply:
Oh no! Well thanks for clarifying :)